packages/react-router-dom/index.tsx

```diff
@@ -133,64 +158,69 @@ if (__DEV__) {
   };
 }
 
+function isModifiedEvent(event: React.MouseEvent) {
+  return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+}
+
 /**
  * The public API for rendering a history-aware <a>.
  */
-export const Link = React.forwardRef(function LinkWithRef(
-  {
-    as: Component = 'a',
-    onClick,
-    replace: replaceProp = false,
-    state,
-    target,
-    to,
-    ...rest
-  },
-  ref
-) {
-  let href = useHref(to);
-  let navigate = useNavigate();
-  let location = useLocation();
-  let toLocation = useResolvedLocation(to);
-
-  function handleClick(event) {
-    if (onClick) onClick(event);
-    if (
-      !event.defaultPrevented && // onClick prevented default
-      event.button === 0 && // Ignore everything but left clicks
-      (!target || target === '_self') && // Let browser handle "target=_blank" etc.
-      !isModifiedEvent(event) // Ignore clicks with modifier keys
-    ) {
-      event.preventDefault();
-
-      let isSameLocation =
-        toLocation.pathname === location.pathname &&
-        toLocation.search === location.search &&
-        toLocation.hash === location.hash;
-
-      // If the pathname, search, and hash haven't changed, a
-      // regular <a> will do a REPLACE instead of a PUSH.
-      let replace = !!replaceProp || isSameLocation;
-
-      navigate(to, { replace, state });
+export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
+  function LinkWithRef(
+    { onClick, replace: replaceProp = false, state, target, to, ...rest },
+    ref
+  ) {
+    let href = useHref(to);
+    let navigate = useNavigate();
+    let location = useLocation();
+    let toLocation = useResolvedLocation(to);
+
+    function handleClick(event: React.MouseEvent<HTMLAnchorElement>) {
+      if (onClick) onClick(event);
+      if (
+        !event.defaultPrevented && // onClick prevented default
+        event.button === 0 && // Ignore everything but left clicks
+        (!target || target === '_self') && // Let browser handle "target=_blank" etc.
+        !isModifiedEvent(event) // Ignore clicks with modifier keys
+      ) {
+        event.preventDefault();
+
+        let isSameLocation =
```

Comment:
Even so it's not directly part of the convertion to TypeScript, I think we should extract this into a separate function. I think it would also be handy for developers in their own projects:

```tsx
function isEqualLocation(a, b): boolean {
  return (
    a.pathname === b.pathname &&
      a.search === b.search &&
      a.hash === b.hash
  );
}

Reply:
I'd prefer to leave it inline for now, thanks. We used to have a function like this, but it wasn't very useful on its own.